ganga/GangaLHCb/Lib/Mergers/GaudiExecMerger.py

```diff
@@ -0,0 +1,253 @@
+                                                   'This can be overridden with the ignorefailed flag.' % (f, j.fqid))
+                # files[f].extend(matchedFiles)
+
+        for k in files.keys():
+            # make sure we are not going to over write anything
+            outputfile = os.path.join(outputdir, k)
+            if os.path.exists(outputfile) and not overwrite:
+                raise PostProcessException('The merge process can not continue as it will result in over writing. '
+                                           'Either move the file %s or set the overwrite flag to True.' % outputfile)
+
+            # make the directory if it does not exist
+            if not os.path.exists(outputdir):
+                os.makedirs(outputdir)
+
+            # recreate structure from output sandbox
+            outputfile_dirname = os.path.dirname(outputfile)
+            if outputfile_dirname != outputdir:
+                if not os.path.exists(outputfile_dirname):
+                    os.mkdir(outputfile_dirname)
+
+            # check that we are merging some files
+            if not files[k]:
+                logger.warning('Attempting to merge with no files. Request will be ignored.')
+                continue
+
+            # check outputfile != inputfile
+            for f in files[k]:
+                if f == outputfile:
+                    raise PostProcessException(
+                        'Output file %s equals input file %s. The merge will fail.' % (outputfile, f))
+            # merge the lists of files with a merge tool into outputfile
+            msg = None
+            try:
+                self.mergefiles(masterjob, files[k], outputfile)
+
+                # create a log file of the merge
+                # we only get to here if the merge_tool ran ok
+                log_file = '%s.merge_summary' % outputfile
+                with open(log_file, 'w') as log:
+                    log.write('# -- List of files merged -- #\n')
+                    for f in files[k]:
+                        log.write('%s\n' % f)
+                    log.write('# -- End of list -- #\n')
+
+            except PostProcessException as e:
+                msg = str(e)
+
+                # store the error msg
+                log_file = '%s.merge_summary' % outputfile
+                with open(log_file, 'w') as log:
+                    log.write('# -- Error in Merge -- #\n')
+                    log.write('\t%s\n' % msg)
+                raise e
+
+        return self.success
+
+
+    def mergefiles(self, masterjob, file_list, output_file):
+
+        #First grab the job object etc.
+        j = masterjob
+        sharedir = j.application.is_prepared.path()
+        tmp_dir = tempfile.gettempdir()
+        #Extract the run script
+        tar = tarfile.open(os.path.join(sharedir, "cmake-input-sandbox.tgz"), "r:gz")
```

Comment:
The temporary file and its directory should be cleaned up after use. This should be done in a way such that it happens if an exception is thrown.